compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -788,3 +788,64 @@ macro_rules! define_queries {
         }
     }
 }
+
+pub fn print_query_stack<'tcx>(
```

Comment:
Should this be an inherent method on `QueryCtxt`?

Reply:
Sure, done : )

Reply:
Edit: I changed to use `rayon::join` now so no need to use `with/enter_context`. Reverted the change to lift up `print_query_stack`.